src/components/digitalSpecimen/tourSteps/masTourSteps.tsx

```diff
@@ -0,0 +1,175 @@
+
+    /* Base variables */
+    const tourTopic = useAppSelector(getTourTopic);
+    const masSteps = DigitalSpecimenTourStepsText.mas;
+    const { options } = StepsConfig();
+    const [steps, setSteps] = useState<{
+        intro: string,
+        element?: string
+    }[]>([]);
+
+    /* Construct Intro.js steps for MAS functionality on the digital specimen page */
+    trigger.SetTrigger(() => {
+        const steps: {
+            intro: string,
+            element?: string
+        }[] = [];
+
+        masSteps.forEach((step, index) => {
+            if ([0, 1, 2, 3, 4, 5, 6].includes(index) || (KeycloakService.IsLoggedIn() && KeycloakService.GetParsedToken()?.orcid)) {
+                steps.push({
+                    intro: step,
+                    element: `.tourMas${index + 1}`
+                });
+            }
+        });
+
+        setSteps(steps);
+    }, []);
+
+    /**
+     * Function that checks what to do on a step change
+     * @param nextIndex The next (selected) index in the step chain
+     * @param resolve Function to resolve the step promise 
+     */
+    const OnStepChange = async (nextIndex: number, resolve: Function) => {
+        /* Handlers for step 1, 2 and 3 */
+        if ([1, 2, 3].includes(nextIndex) && annotationMode) {
+            SetAnnotationMode(false);
+        } else if (![1, 2, 3].includes(nextIndex) && !annotationMode) {
+            SetAnnotationMode(true);
+        }
+
+        if ([5, 6, 7, 8, 9, 10, 11, 12, 13].includes(nextIndex)) {
+            dispatch(setMasMenuToggle(true));
+        } else {
+            dispatch(setMasMenuToggle(false));
+        }
+
+        if ([8, 9, 10].includes(nextIndex)) {
+            dispatch(setMasScheduleMenuToggle(true));
+        } else {
+            dispatch(setMasScheduleMenuToggle(false));
+        }
+
+        if ([9, 10].includes(nextIndex)) {
+            dispatch(setMasDummy({
+                '@id': 'MachineAnnotationServiceDummy',
+                '@type': 'ods:MachineAnnotationService',
+                'schema:identifier': 'MachineAnnotationServiceDummy',
+                'ods:fdoType': 'https://doi.org/10.15468/1a2b3c',
+                'schema:name': 'Dummy Machine Annotation Service',
+                'schema:description': 'A machine annotation service for showing off functionality',
+                'schema:dateCreated': '2024-11-15T08:56:50.758Z',
+                'schema:creator': {
+                    '@id': 'MachineAnnotationServiceDummyAgent',
+                    '@type': 'prov:SoftwareAgent'
+                },
+                'schema:dateModified': '2024-11-15T08:56:50.758Z',
+                'ods:containerImage': 'dummyContainerImage',
+                'ods:containerTag': 'dummyContainerTag',
+                'ods:batchingPermitted': false,
+                'ods:timeToLive': 1000
+            }));
+        } else {
+            dispatch(setMasDummy(undefined));
+        }
+
+        if ([11, 12, 13].includes(nextIndex)) {
```

Comment:
Something like this might make it easier to pick out the correct states and simplify `if` logic
```
const stepConfigs = {
    masMenuSteps: [5, 6, 7, 8, 9, 10, 11, 12, 13],
    masScheduleMenuSteps: [8, 9, 10],
    masDummySteps: [9, 10],
    masJobRecordSteps: {
        steps: [11, 12, 13],
        states: { 11: 'SCHEDULED', 12: 'COMPLETED', 13: 'FAILED' }
    },
};
```

Reply:
Fixed